### PR TITLE
CIV-7788 Hearing fee look up

### PR DESCRIPTION
**Pull request by miguelMolina3691**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-7788

### Change description ###
Look up hearing fee

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/service/hearings/HearingFeesServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        given(restTemplate.getForObject(queryCaptor.capture(), eq(FeeLookupResponseDto.class)))
+            .willReturn(FeeLookupResponseDto.builder()
+                            .code("test_fee_code")
+                            .version(1)
+                            .build());
+
+        BigDecimal claimAmount = new BigDecimal(125);
+        // When
+        Exception exception = assertThrows(
+            RestClientException.class, () -> feesService.getFeeForHearingSmallClaims(claimAmount));
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/hearings/HearingFeesServiceTest.java:150]: The following is needed if you apply other suggestions.

```suggestion
            InternalServerErrorException.class, () -> feesService.getFeeForHearingSmallClaims(claimAmount));
```

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
Some improvements can be made to the code quality.

Once you get to it, if you apply my suggestions make sure to run the build locally as I expect at least one checkstyle exception and one compile exception due to imports.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/HearingScheduledHandler.java [hunk 2]
```diff
@@ -179,32 +183,28 @@ private CallbackResponse handleAboutToSubmit(CallbackParams callbackParams) {
...
+            claimAmount = caseData.getClaimValue().toPounds();
+        } else {
+            if (nonNull(caseData.getTotalInterest())) {
+                claimAmount = caseData.getTotalClaimAmount()
+                    .add(caseData.getTotalInterest())
+                    .setScale(2, RoundingMode.UNNECESSARY);
+            } else {
+                claimAmount = caseData.getTotalClaimAmount().setScale(2, RoundingMode.UNNECESSARY);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (miguelMolina3691) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/HearingScheduledHandler.java:203]: Modified

File: src/main/java/uk/gov/hmcts/reform/civil/service/hearings/HearingFeesService.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            .queryParam(CHANNEL, feesConfiguration.getChannel())
+            .queryParam(EVENT, feesConfiguration.getHearingEvent())
+            .queryParam(JURISDICTION1, feesConfiguration.getJurisdiction1())
+            .queryParam(JURISDICTION2, feesConfiguration.getJurisdiction2Hearing())
+            .queryParam(SERVICE, feesConfiguration.getService())
+            .queryParam(KEYWORD, feesConfiguration.getFastTrackHrgKey())
+            .queryParam(AMOUNT, amount);
+
+        return getRespond(builder);
+    }
```
[the remark this answers is not in the document]
Comment (miguelMolina3691) [reply src/main/java/uk/gov/hmcts/reform/civil/service/hearings/HearingFeesService.java:62]: This is due to the fees registry. There is a jurisdiction field and a jurisdiction2 field, which for some fees is different

**Pull request merged**